packages/block-library/src/block/index.php

```diff
@@ -41,13 +41,31 @@ function render_block_core_block( $attributes ) {
 
 	$seen_refs[ $attributes['ref'] ] = true;
 
+	$filter_block_context = static function( $context ) use ( $attributes ) {
```

Comment:
The filtering only makes sense if `$attributes['dynamicContent']` is provided, so maybe it should be created conditionally. 

Should the filter get applied only around: `$content = do_blocks( $content );`? It might help to reason where it takes an effect.